beacon-chain/sync/rpc_beacon_blocks_by_range.go

```diff
@@ -214,6 +217,37 @@ func (s *Service) validateRangeRequest(r *pb.BeaconBlocksByRangeRequest) error {
 	return nil
 }
 
+// filters all the provided blocks to ensure they are canonical
+// and are strictly linear.
+func (s *Service) filterBlocks(ctx context.Context, blks []*ethpb.SignedBeaconBlock, roots [][32]byte, prevRoot *[32]byte,
+	step, startSlot uint64) ([]*ethpb.SignedBeaconBlock, error) {
+
```

Comment:
perhaps do a defensive check that `blks` and `roots` are the same length

Reply:
This is actually already done in `dedupBlocksAndRoots`  but I can add it in again too.